ui/apps/platform/src/Containers/Search/SearchResults.tsx

```diff
@@ -195,20 +200,44 @@ function SearchResults({
         setGlobalSearchCategory(selectedTab.category);
     }
 
+    const amendSearchOptions = (searchCategory: string, category: string, name: string) => {
+        let searchOptions = globalSearchOptions.slice();
```

Comment:
While we're refactoring, I think [...spread] is a clearer idiom for copying. Or a end-line comment. Or both.

(JSPerf actually shows slice as slightly faster than spread for 3.5 million elements, but probably not worth worrying about in this case.)

Reply:
Especially if these are the only calls of the recently added `add` functions (pardon pun) what do you think about:
* Rewrite `add` function as `get` function
* In block of `if` statement, `return [...globalSearchOptions, getSearchModifier(…), getSearchKeyword(…)];`
* Otherwise, function returns `globalSearchOptions`

Reply:
Since this was the only occurrence of those two functions, and we are moving away from the "search option" data structure, and those functions were written 4 years ago, I just in-lined the functionality here.

I suspect the `className: 'Select-create-option-placeholder'` piece is outdated as well, since it only occurs in one other place in the code from four years ago. I haven't removed it though, as it is possible it is referenced by some CSS elsewhere that I can't search for. (Removing it would get rid of the cast to `GlobalSearchOption` though...)